app/src/main/java/org/onionshare/android/files/FileManager.kt

```diff
@@ -113,15 +113,16 @@ class FileManager @Inject constructor(
     suspend fun zipFiles() = withContext(Dispatchers.IO) {
         if (state.value is Zipped) return@withContext
         zipJob?.cancelAndJoin()
-        try {
-            @Suppress("BlockingMethodInNonBlockingContext")
-            zipJob = launch { zipFilesInternal() }
-        } catch (e: FileErrorException) {
-            // remove errorFile from list of files, so user can try again
-            val newFiles = state.value.files.toMutableList().apply { remove(e.file) }
-            _state.value = FilesState.Error(newFiles, e.file)
-        } catch (e: IOException) {
-            _state.value = FilesState.Error(state.value.files)
+        zipJob = launch {
+            try {
+                zipFilesInternal()
```

Comment:
That's the main issue that showed as a crash: When launching a new coroutine in the current coroutine context, the exception thrown in there would bubble up to the parent context, so our try/catch wasn't actually catching it :facepalm: